clusterman/draining/queue.py

```diff
@@ -422,33 +430,37 @@ def process_warning_queue(self) -> bool:
             autoscaling_resource_groups: Set[str] = set()
             # we do this in two loops since we only use SFRs for Mesos, but we use Spot
             # ASGs for Kubernetes
-            for pool in get_pool_name_list(self.cluster, "mesos"):
-                pool_config = staticconf.NamespaceReaders(POOL_NAMESPACE.format(pool=pool, scheduler="mesos"))
-                for resource_group_conf in pool_config.read_list("resource_groups"):
-                    spot_fleet_resource_groups |= set(
-                        SpotFleetResourceGroup.load(
-                            cluster=self.cluster,
-                            pool=pool,
-                            config=list(resource_group_conf.values())[0],
-                        ).keys()
-                    )
-
-            for pool in get_pool_name_list(self.cluster, "kubernetes"):
-                pool_config = staticconf.NamespaceReaders(POOL_NAMESPACE.format(pool=pool, scheduler="kubernetes"))
-                for resource_group_conf in pool_config.read_list("resource_groups"):
-                    autoscaling_resource_groups |= set(
-                        AutoScalingResourceGroup.load(
-                            cluster=self.cluster,
-                            pool=pool,
-                            config=list(resource_group_conf.values())[0],
-                        ).keys()
-                    )
-
+            if arrow.now() > self.resource_group_cache_expiration_time:
+
+                for pool in get_pool_name_list(self.cluster, "mesos"):
+                    pool_config = staticconf.NamespaceReaders(POOL_NAMESPACE.format(pool=pool, scheduler="mesos"))
+                    for resource_group_conf in pool_config.read_list("resource_groups"):
+                        spot_fleet_resource_groups |= set(
+                            SpotFleetResourceGroup.load(
+                                cluster=self.cluster,
+                                pool=pool,
+                                config=list(resource_group_conf.values())[0],
+                            ).keys()
+                        )
+                self.spot_fleet_resource_groups_cache = spot_fleet_resource_groups
```

Comment:
Another optimization you could make is to pre-match the group_id you get for `host_to_process`, as I guess names for SFR will have a somewhat stable prefix (I think it's `sfr-` and some hex chars, but I'd have to verify), so if something "looks like" an SFR host, you just load SFR IDs, and if not you just load ASGs names.

Reply:
Probably this code will be removed soon. Additionally, I think it's little bit dangerous because someone can change this logic (pre-match) without notify us